spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java

```diff
@@ -206,20 +213,20 @@ public MetricStore metricStore(MetricsProperties metricsProperties) {
 	}
 
 	@Bean
-	@ConditionalOnBean(StreamDefinitionRepository.class)
+	@ConditionalOnStreamsEnabled
 	public StreamDefinitionController streamDefinitionController(StreamDefinitionRepository repository,
 			StreamService streamService) {
 		return new StreamDefinitionController(streamService);
 	}
 
 	@Bean
-	@ConditionalOnBean(StreamDefinitionRepository.class)
+	@ConditionalOnStreamsEnabled
 	public StreamValidationController streamValidationController(StreamService streamService) {
 		return new StreamValidationController(streamService);
 	}
 
 	@Bean
-	@ConditionalOnBean(TaskDefinitionRepository.class)
+	@ConditionalOnTasksEnabled
```

Comment:
Consider grouping all ConditionalOnTasksEnabled beans in a common inner configuration class.